### PR TITLE
feat/pact generator lexer only

### PR DESCRIPTION
**Pull request by alber70g**
- chore(monorepo): adds eslint to all packages for IDE support
- feat(pactjs-generator): adds lexer + parser for module and funcs

### CODE REVIEW

File: packages/libs/pactjs-generator/src/lexer.ts [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+
+/**
+ * @internal
+ */
+export const lexer: Lexer = moo.compile({
+  TYPED_IDEN: /[a-zA-Z-_0-9]+:[a-zA-Z-_{}.0-9]+/,
+  name: {
+    match: /[a-zA-Z-_*]+/,
+    type: moo.keywords({
+      KW_BLESS: 'bless',
```
Comment (Randynamic) [inline packages/libs/pactjs-generator/src/lexer.ts:12]: @ggobugi27 Can you check if we cover all keywords here?

File: packages/libs/pactjs-generator/src/lexer.ts [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+  let token: moo.Token | undefined = undefined;
+  const state: IState = { stack: [], moduleName: '', defunName: '' } as IState;
+
+  const output: Output = {} as Output;
+  while ((token = lexer.next())) {
+    logger(token);
+    switch (token.type) {
+      case 'KW_MODULE':
+        state.stack.push('module');
+        logger('push module');
```
Comment (Randynamic) [inline packages/libs/pactjs-generator/src/lexer.ts:129]: Shouldn't we also log `token.type` here to get more specific logs?

File: packages/libs/pactjs/src/index.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
+import { Pact } from './pact';
```
Comment (Randynamic) [inline packages/libs/pactjs/src/index.ts:1]: This is for testing purposes, right?

**Review by Randynamic — DISMISSED** [review]

**Review by Randynamic — DISMISSED** [review]

**Review by Randynamic — DISMISSED** [review]

File: packages/libs/pactjs-generator/src/lexer.ts (hunk 2 above)
Comment (alber70g) [reply packages/libs/pactjs-generator/src/lexer.ts:111]: It's logging the full object at the start of the `while` loop

File: packages/libs/pactjs/src/index.ts (hunk 3 above)
Comment (alber70g) [reply packages/libs/pactjs/src/index.ts:1]: Yes. The `index.ts` is an outline of what the interface could look like

**Comment by Randynamic** [comment]
@alber70g can you check why the build is still failing so we can get this PR in.

**Comment by alber70g** [comment]
Closing in favor of #59

**Comment by alber70g** [comment]
> @alber70g can you check why the build is still failing so we can get this PR in.

It's draft. Far from complete 😉

**Pull request closed without merging**